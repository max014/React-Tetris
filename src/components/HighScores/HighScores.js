import React, { Component } from 'react';
import styles from './HighScores.module.css';
import { connect } from 'react-redux';
import * as actions from '../../store/actions/scoreBoard';

export class HighScores extends Component {

	componentDidMount() {
		this.props.getScores();
	}

	render() {
		return (
			<div className={styles.HighScores}>
				<h3>High Scores</h3>
				{this.props.scores.map((score, index) => (
					<div key={index} className={styles.container}>
						<h4>{score.name + " - " + score.score}</h4>
					</div>
				))}
			</div>
		);
	}
}

const mapStateToProps = state => {
    return {
    	scores: state.scores
    };
}

const mapDispatchToProps = dispatch => {
    return {
        getScores: () => dispatch(actions.getScores())
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(HighScores);