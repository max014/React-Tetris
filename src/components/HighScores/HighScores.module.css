.HighScores{
	position: absolute;
	left: 5%;
	top: 5%;
	width: 20vw;
	background: linear-gradient(to right, #111, #222);
	padding-left: 20px;
	border: 4px solid white;
}

.HighScores h3,
.HighScores h4{
	color: white;
}