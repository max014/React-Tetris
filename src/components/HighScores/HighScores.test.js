import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HighScores } from './HighScores';

describe('HighScores', () => {
    const mockgetScores = jest.fn();
    const props = {
        scores: [
            {"_id":"1","name":"Samsquanch","score":416800},
            {"_id":"2","name":"Samsquanch","score":416800},
            {"_id":"3","name":"Samsquanch","score":416800},
        ],
        getScores: mockgetScores
    }
    const highScores = shallow(<HighScores {...props}/>);
    it('renders properly', () => {
        expect(toJson(highScores)).toMatchSnapshot();
    });
    it('calls `getScores` prop when initialized', () => {
        expect(mockgetScores).toHaveBeenCalled();
    });
});