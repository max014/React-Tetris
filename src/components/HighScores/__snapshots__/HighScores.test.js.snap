// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighScores renders properly 1`] = `
<div
  className="HighScores"
>
  <h3>
    High Scores
  </h3>
  <div
    className="container"
    key="0"
  >
    <h4>
      Samsquanch - 416800
    </h4>
  </div>
  <div
    className="container"
    key="1"
  >
    <h4>
      Samsquanch - 416800
    </h4>
  </div>
  <div
    className="container"
    key="2"
  >
    <h4>
      Samsquanch - 416800
    </h4>
  </div>
</div>
`;
